### PR TITLE
Automatically update binary hashes

### PR DESCRIPTION
**Pull request by variadico**
This change does the following:

* deletes `update-server-version.sh` and moves all that functionality into `update.py`.
* adds new functionality to automatically update the server binary hashes by using the SHASUM file from the releases page.
* tweaks the existing Dockerfile a little, mostly just small refactoring
* updates readme to reflect changes

For testing, first I downgraded NATS to 2.2.5 (8dcdf68). Then I used `update.py` to upgrade back to 2.2.6 (c0b580c). Once we're happy with the change, we can squash the commits down.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,38 +25,33 @@ For the most part, image variant Dockerfiles will download the official NATS
...
-```
+## Manual updates
 
-This script doesn't update everything though. Here are some other things you
+This script doesn't update everything. Here are some other things you
 may or may not want to update.
 
 * The Ubuntu host version used for CI.
 * The Windows host versions used for CI.
+* The Alpine version
```
Comment (kozlovic) [inline README.md:54]: Yes, currently one has to update the directory name, then the version in the Dockerfile itself. Now, there is reference to alpine directory name in the update.py script.
Maybe we can have it done in a different PR ;-)

**Review by kozlovic — APPROVED** [review]
LGTM! Thank you, this is great and will save me some work when doing the release. As a stretch goal it would be nice if the update process could optionally change the alpine version.

You can squash the changes since the experimentation of going down to 2.2.5 and back up to 2.2.6 works (I have tested locally myself too).

**Comment by variadico** [comment]
All right. I squashed down all the commits.

I'll submit a future PR to update the Alpine image. In addition to updating Alpine, I'm guessing we also probably want to update the Windows images at some point, right?

**Comment by variadico** [comment]
Merging.

**Comment by kozlovic** [comment]
> I'm guessing we also probably want to update the Windows images at some point, right?

I think that currently those are the only one that are supported. See conversation following the comment I posted [there](https://github.com/docker-library/official-images/pull/9991#issuecomment-829481429)

**Comment by variadico** [comment]
Ooooh, ok.

> ... we cannot build anything other than ltsc2016 and 1809...

https://github.com/docker-library/official-images/issues/9198#issuecomment-737477765
